error[E0728]: `await` is only allowed inside `async` functions and blocks
  --> $DIR/issue-62009-1.rs:7:5
   |
LL | fn main() {
   |    ---- this is not `async`
LL |     async { let (); }.await;
   |     ^^^^^^^^^^^^^^^^^^^^^^^ only allowed inside `async` functions and blocks

error[E0728]: `await` is only allowed inside `async` functions and blocks
  --> $DIR/issue-62009-1.rs:9:5
   |
LL |   fn main() {
   |      ---- this is not `async`
...
LL | /     async {
LL | |
LL | |         let task1 = print_dur().await;
LL | |     }.await;
   | |___________^ only allowed inside `async` functions and blocks

error[E0728]: `await` is only allowed inside `async` functions and blocks
  --> $DIR/issue-62009-1.rs:13:5
   |
LL | fn main() {
   |    ---- this is not `async`
...
LL |     (|_| 2333).await;
   |     ^^^^^^^^^^^^^^^^ only allowed inside `async` functions and blocks

error[E0277]: the trait bound `[closure@$DIR/issue-62009-1.rs:13:5: 13:15]: std::future::Future` is not satisfied
  --> $DIR/issue-62009-1.rs:13:5
   |
LL |     (|_| 2333).await;
   |     ^^^^^^^^^^^^^^^^ the trait `std::future::Future` is not implemented for `[closure@$DIR/issue-62009-1.rs:13:5: 13:15]`
   | 
  ::: $SRC_DIR/libstd/future.rs:LL:COL
   |
LL |     F: Future
   |        ------ required by this bound in `std::future::poll_with_tls_context`

error: aborting due to 4 previous errors

Some errors have detailed explanations: E0277, E0728.
For more information about an error, try `rustc --explain E0277`.
